### ISSUE TITLE
Game aborts, when River (II) can not be placed in full

### ISSUE THREAD

**Issue by Hounk** [farin/JCloisterZone#155]
![_ai_20150712-125023](https://cloud.githubusercontent.com/assets/12594204/8637539/a40a185c-2896-11e5-896d-3fa6f4b33753.png)

According to CAR 7.4, p. 81f. (although this includes the count, it should apply to any other situation as well), the rest of the river tiles should just be discarded and the game continue, even if this means one or both river branches end without lake. This follows the basic rule, that in the rare occasions, a tile is drawn, which can not be placed, it is discarded.

**Comment by farin** [comment]
yes, that is old one https://github.com/farin/JCloisterZone/issues/34

**Comment by Hounk** [comment]
Quote TexGG in the other issue: Seem the AI miss one rule limitation, since player can not do that.

That's not really true in this case, because I forced the abortion of the river myself, placing the tile with "4". However, it was not a placement violating the rules, because they prohibit just a straight u-turn (+ the connection of the two branches to a loop). However, this rule makes it (with the 2 branches of River II, River I is "fail safe") not impossible, just less likely, that the river can not be placed in full.

In summary, not the placement of the tiles themselves violates the official rules, just the fact, that the game is discontinued, when the river can not be placed completely, rather than that the rest of the river tiles (or the ones, which can not be placed) are discarded (like for example a CCCC, when there is no free city-edge for it) and the game continues with the rest of the tiles.
